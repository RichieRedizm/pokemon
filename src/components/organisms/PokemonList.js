import React, { Fragment, useContext } from 'react'
import PropTypes from 'prop-types'
import GraphQLContext from '../../context/graphQL/graphQLContext'
import PokemonListItem from '../molecules/PokemonListItem'
import Spinner from '../search/Spinner'
import isEmpty from 'lodash/isEmpty'

const PokemonList = () => {
  const graphQLContext = useContext(GraphQLContext)
  const { loading, pokemons } = graphQLContext

  if (loading) return <Spinner />

  return (
    <Fragment>
      {!isEmpty(pokemons) && (
        <div data-testid='pokemon-list' className='grid-245'>
          {pokemons.map(pokemon => (
            <PokemonListItem key={pokemon.id} pokemon={pokemon} />
          ))}
        </div>
      )}
    </Fragment>
  )
}

PokemonList.contextTypes = {
  pokemons: PropTypes.array,
  loading: PropTypes.bool
}

export default PokemonList
