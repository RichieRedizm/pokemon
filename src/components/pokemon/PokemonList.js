import React from 'react'

const PokemonList = () => {
  return (
    <div className='card text-center'>
      <h3>pokemon name</h3>
    </div>
  )
}

export default PokemonList
