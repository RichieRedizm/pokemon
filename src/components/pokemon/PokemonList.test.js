import React from 'react'
import GraphQLContext from '../../context/graphQL/graphQLContext'
import { render, cleanup } from '@testing-library/react'
import PokemonList from './PokemonList'

afterEach(cleanup)

function renderPokemonList(state) {
  return render(
    <GraphQLContext.Provider value={state}>
      <PokemonList />
    </GraphQLContext.Provider>
  )
}
test('if loading set to true load Spinner component', () => {
  const { getByTestId, asFragment } = renderPokemonList({
    pokemons: [],
    loading: true
  })

  expect(getByTestId('loading'))
  expect(asFragment()).toMatchSnapshot()
})

test('if pokemons data not null generate list of pokemon items', () => {
  const { getByTestId, asFragment } = renderPokemonList({
    pokemons: [
      {
        id: 'UG9rZW1vbjowMDE=',
        number: '001',
        name: 'Bulbasaur',
        image: 'https://img.pokemondb.net/artwork/bulbasaur.jpg',
        __typename: 'Pokemon'
      },
      {
        id: 'UG9rZW1vbjowMDI=',
        number: '002',
        name: 'Ivysaur',
        image: 'https://img.pokemondb.net/artwork/ivysaur.jpg',
        __typename: 'Pokemon'
      }
    ],
    loading: false
  })
  const pokemonList = getByTestId('pokemon-list')
  expect(pokemonList)
  expect(pokemonList.children.length).toBe(2)
  expect(asFragment()).toMatchSnapshot()
})

test('if pokemons array empty dont display any list items', () => {
  const { queryByTestId } = renderPokemonList({
    pokemons: null,
    loading: false
  })
  expect(queryByTestId('pokemon-list')).toBeNull()
})
