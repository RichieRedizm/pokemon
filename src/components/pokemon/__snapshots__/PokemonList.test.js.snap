// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if loading set to true load Spinner component 1`] = `
<DocumentFragment>
  <img
    alt="loading"
    data-testid="loading"
    src="spinner.gif"
    style="width: 200px; margin: auto; display: block;"
  />
</DocumentFragment>
`;

exports[`if pokemons data not null generate list of pokemon items 1`] = `
<DocumentFragment>
  <div
    data-testid="pokemon-list"
    style="display: grid; grid-template-columns: repeat(5, 1fr);"
  >
    <div
      class="card text-center pokemon-item-card"
    >
      <h3>
        Bulbasaur
      </h3>
    </div>
    <div
      class="card text-center pokemon-item-card"
    >
      <h3>
        Ivysaur
      </h3>
    </div>
  </div>
</DocumentFragment>
`;
